static/aggregate_test.go

```diff
@@ -0,0 +1,199 @@
+package static_test
+
+import (
+	"github.com/dogmatiq/configkit"
+	"github.com/dogmatiq/configkit/message"
+	. "github.com/dogmatiq/configkit/static"
+	"github.com/dogmatiq/dogma/fixtures"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"golang.org/x/tools/go/packages"
+)
+
+var _ = Describe("func FromPackages() (aggregate analysis)", func() {
+	When("the application contains a single aggregate handler", func() {
+		It("returns the aggregate handler configuration", func() {
+			cfg := packages.Config{
+				Mode: packages.LoadAllSyntax,
+				Dir:  "testdata/aggregates/single-aggregate-app",
+			}
+
+			pkgs, err := packages.Load(&cfg, "./...")
+			Expect(err).NotTo(HaveOccurred())
+
+			apps := FromPackages(pkgs)
+			Expect(apps).To(HaveLen(1))
+			Expect(apps[0].Handlers().Aggregates()).To(HaveLen(1))
+
+			a := apps[0].Handlers().Aggregates()[0]
+			Expect(a.Identity()).To(
+				Equal(
+					configkit.Identity{
+						Name: "<aggregate>",
+						Key:  "ef16c9d1-d7b6-4c99-a0e7-a59218e544fc",
+					},
+				),
+			)
+			Expect(a.TypeName()).To(
+				Equal(
+					"github.com/dogmatiq/configkit/static/testdata/aggregates/single-aggregate-app.AggregateHandler",
+				),
+			)
+			Expect(a.HandlerType()).To(Equal(configkit.AggregateHandlerType))
+			Expect(
+				a.MessageNames().Consumed.IsEqual(
+					message.NameRoles{
+						message.NameOf(fixtures.MessageA{}): message.CommandRole,
+						message.NameOf(fixtures.MessageB{}): message.CommandRole,
+					},
+				),
+			).To(BeTrue())
+			Expect(
+				a.MessageNames().Produced.IsEqual(
+					message.NameRoles{
+						message.NameOf(fixtures.MessageC{}): message.EventRole,
+						message.NameOf(fixtures.MessageD{}): message.EventRole,
+					},
+				),
+			).To(BeTrue())
+		})
+	})
+
+	When("the application contains multiple aggregate handlers", func() {
+		It("returns all of the aggregate handler configurations", func() {
+			cfg := packages.Config{
+				Mode: packages.LoadAllSyntax,
+				Dir:  "testdata/aggregates/multiple-aggregate-app",
+			}
+
+			pkgs, err := packages.Load(&cfg, "./...")
+			Expect(err).NotTo(HaveOccurred())
+
+			apps := FromPackages(pkgs)
+			Expect(apps).To(HaveLen(1))
+			Expect(apps[0].Handlers().Aggregates()).To(HaveLen(2))
+
+			a1 := apps[0].Handlers().Aggregates()[0]
```

Comment:
This is relying on the fact that `ssautil.AllPackages()` function parses files within a package in  alphabetic order . To me, there is a concern that this can potentially make the tests fragile in case the logic within `ssautil.AllPackages()` changes.

Reply:
To cope with this problem I am proposing the approach highlighted in https://github.com/dogmatiq/configkit/pull/123/commits/9614e85ff4777a5414e9b4dbf5c64555662f320d.  IMHO, using `configkit.HandlerSet.ByIdentity()` simplifies the aggregate retrieval and yet we still test its remaining components regardless of the aggregate order in `configkit.HandlerSet.Aggregates()`.

Reply:
A slice of aggregate identities is now used to test multiple aggregate handlers in https://github.com/dogmatiq/configkit/pull/123/commits/b2bd8a448a89729750a8c1eaba730119f297f439.